src/Arcus.Security.Tests.Integration/KeyVault/KeyVaultSecretProvider.ServicePrincipalTests.cs

```diff
@@ -0,0 +1,325 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Arcus.Security.Core;
+using Arcus.Security.Core.Caching.Configuration;
+using Arcus.Security.Providers.AzureKeyVault;
+using Arcus.Security.Providers.AzureKeyVault.Configuration;
+using Azure;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+using Xunit;
+
+namespace Arcus.Security.Tests.Integration.KeyVault
+{
+    [Trait(name: "Category", value: "Integration")]
+    public partial class KeyVaultSecretProviderTests
```

Comment:
Moved the existing tests to dedicated 'managed identity' and 'service principal' tests.
For the 'certificate tests', I created an issue: #425